source/simulator/stokes_matrix_free.cc

```diff
@@ -984,8 +981,7 @@ namespace aspect
 
     for (unsigned int cell=cell_range.first; cell<cell_range.second; ++cell)
       {
-        AssertDimension(viscosity_x_2.size(0), data.n_macro_cells());
-        AssertDimension(viscosity_x_2.size(1), velocity.n_q_points);
+        VectorizedArray<number> cell_viscosity_x_2 = viscosity_x_2(cell);
```

Comment:
do you need to make a copy here? I would think a const ref would be enough

Reply:
Is having a const ref in the cell loop better than looking up the value each time in the quadrature loop below (where the value is used)?

Reply:
I don't know to be honest and it is hard to figure out without looking at assembly.